Formula/g/gpgme.rb

```diff
@@ -37,20 +38,43 @@ def install
     # hardcoded, the Arch Linux patch that changed 3.9 to 3.10 can't detect 3.11
     inreplace "configure", /# Reset everything.*\n\s*unset PYTHON$/, ""
 
-    # Uses generic lambdas.
-    # error: 'auto' not allowed in lambda parameter
-    ENV.append "CXXFLAGS", "-std=c++14"
-
     # Use pip over executing setup.py, which installs a deprecated egg distribution
     # https://dev.gnupg.org/T6784
     inreplace "lang/python/Makefile.in",
               /^\s*\$\$PYTHON setup\.py\s*\\/,
               "$$PYTHON -m pip install --use-pep517 #{std_pip_args.join(" ")} . && : \\"
 
+    # Required for Qt 6 bindings
+    ENV.append "CXXFLAGS", "-std=c++17"
+
+    ENV["PKG_CONFIG_PATH"] = [
+      "#{Formula["qt@6"].libexec}/lib/pkgconfig",
+      "#{HOMEBREW_PREFIX}/lib/pkgconfig",
+    ].join(":")
+
     system "./configure", "--disable-silent-rules",
                           "--enable-static",
+                          "--enable-languages=cl,cpp,python,qt6",
                           *std_configure_args
-    system "make"
+
+    # Required to make Qt dependency effectively optional and to pass `brew linkage --strict` test
+    if OS.mac?
+      inreplace "lang/qt/src/Makefile",
+                "-framework QtCore",
+                "-Wl,-weak_framework,QtCore"
+    end
+    if OS.linux?
+      inreplace "lang/qt/src/Makefile",
+                "-lQt6Core",
+                "-weak Qt6Core"
+    end
+
+    system "make", "SHELL=/bin/bash"
+    #              ^^^
+    # A workaround for an issue introduced around 1.24.0 upstream.
+    # On macOS, `echo` in `/bin/sh` is unaware of `-n` switch (no new line),
+    # so literal `-n` is written into C++ headers, breaking build
```

Comment:
is any issue tracking this?

Reply:
Not yet
a) Discovered just yesterday.
b) macOS-specific due to the very old Bash as `/bin/sh`
I know how to send patches to GnuPG, but it will take time. Lots of time.

Reply:
it's fine if it takes time, as long as upstream is aware of the issue.

Reply:
> it's fine if it takes time, as long as upstream is aware of the issue.

The upstream doesn't care about Bash 3.2 in general and its POSIX-compliant mode in particular.
They focus on Linux and Windows with modern base system software.
No wonder why almost every single patch release of GnuPG requires some build fixes.

At this point, it's us (i.e. macOS users) who have an inadequate build environment -- and this change brings us closer to parity with Linux tools.

Rejecting version upgrade on the basis that I configure build environment for the code seems unreasonable and excessive. It's kind of purpose of this project.
I do not patch the upstream software; I build it in an environment a bit closer to the intended one.

As for the upstream patches, there is a way to fix what's broken with a POSIX-compliant one-liner.
This is one of the two offending places: https://github.com/gpg/gpgme/blob/f6257431a8f80b507b8197f16088fb257153fe4e/lang/qt/src/Makefile.am#L442 #
```diff
$(camelcase_headers): Makefile.am
-	echo -n "#include \"qgpgme/" > "$@"
-	echo -n "$@" | tr '[:upper:]' '[:lower:]' >> "$@"
-	echo ".h\"" >> "$@"
+	echo "#include \"qgpgme/$(shell tr '[:upper:]' '[:lower:]' <<<"$@").h\"" > "$@"
```

Where `<<<` is actually standard-compliant.
```
sh-3.2$ set -o posix
sh-3.2$ echo $POSIXLY_CORRECT 
y
sh-3.2$ tr '[:upper:]' '[:lower:]' <<<QwErTy
qwerty
sh-3.2$ exit
```

Selling such a fix would be a show. Because it's not about functionality, but about code style and supporting broken environments. (Again, it's macOS that is broken.)
In this organization, it would be frankly ridiculous.

Besides, they have disabled self-service registration for their bug tracker, see: https://dev.gnupg.org/auth/start/
Now they vet new users manually. And I happen to have an account approved personally by Werner Koch (the principal author): https://dev.gnupg.org/people/manage/120330/
They are very serious.

Could you please not block this upgrade on a technicality with people who don't even have an open-door policy in their community?

Reply:
This is all reasonable to me.